core/pkg/sync/blob/blob_sync.go

```diff
@@ -0,0 +1,131 @@
+package blob
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/open-feature/flagd/core/pkg/logger"
+	"github.com/open-feature/flagd/core/pkg/sync"
+	"gocloud.dev/blob"
+	_ "gocloud.dev/blob/gcsblob" // needed to initialize GCS driver
+)
+
+type Sync struct {
+	Bucket      string
+	Object      string
+	BlobURLMux  *blob.URLMux
+	Cron        Cron
+	Logger      *logger.Logger
+	Interval    uint32
+	ready       bool
+	lastUpdated time.Time
+}
+
+// Cron defines the behaviour required of a cron
+type Cron interface {
+	AddFunc(spec string, cmd func()) error
+	Start()
+	Stop()
+}
+
+func (hs *Sync) Init(_ context.Context) error {
+	if hs.Bucket == "" {
+		return errors.New("no bucket string set")
+	}
+	if hs.Object == "" {
+		return errors.New("no object string set")
+	}
+	return nil
+}
+
+func (hs *Sync) IsReady() bool {
+	return hs.ready
+}
+
+func (hs *Sync) Sync(ctx context.Context, dataSync chan<- sync.DataSync) error {
+	hs.Logger.Info(fmt.Sprintf("starting sync from %s/%s with interval %d", hs.Bucket, hs.Object, hs.Interval))
+	_ = hs.Cron.AddFunc(fmt.Sprintf("*/%d * * * *", hs.Interval), func() {
+		err := hs.sync(ctx, dataSync, false)
+		if err != nil {
+			hs.Logger.Warn(fmt.Sprintf("sync failed: %v", err))
+		}
+	})
+	// Initial fetch
+	hs.Logger.Debug(fmt.Sprintf("initial sync of the %s/%s", hs.Bucket, hs.Object))
+	err := hs.sync(ctx, dataSync, false)
+	if err != nil {
+		return err
+	}
+
+	hs.ready = true
+	hs.Cron.Start()
+	<-ctx.Done()
+	hs.Cron.Stop()
+
+	return nil
+}
+
+func (hs *Sync) ReSync(ctx context.Context, dataSync chan<- sync.DataSync) error {
+	return hs.sync(ctx, dataSync, true)
+}
+
+func (hs *Sync) sync(ctx context.Context, dataSync chan<- sync.DataSync, skipCheckingModTime bool) error {
+	bucket, err := hs.getBucket(ctx)
+	if err != nil {
+		return fmt.Errorf("couldn't get bucket: %v", err)
+	}
+	defer bucket.Close()
+	var updated time.Time
+	if !skipCheckingModTime {
+		updated, err = hs.fetchObjectModificationTime(ctx, bucket)
+		if err != nil {
+			return fmt.Errorf("couldn't get object attributes: %v", err)
+		}
+		if hs.lastUpdated == updated {
```

Comment:
Optional: Perhaps it's worth logging a warning if hs.lastUpdated < updated and also not update the config in such case.

Reply:
I'm not sure. I wanted to avoid downloading the whole blob every time.
Regarding updated time - GCS updates it automatically on any change so it should always grow.
The other system - I don't know.

However even if there is a change that decreased the time - IMO we should still switch configuration to that file. E.g. if some system would keep the time from the local file and there is a roll-back to old config and the updated time decreases - the flagd should switch to it.

Reply:
Switching config to that file SGTM. But still logging a warning in such case may be useful.

Reply:
Added